adoc/admin-gpus.adoc

```diff
@@ -0,0 +1,163 @@
+
+== Prepare the host machine
+
+=== Install the GPU drivers
+
+Not every worker node in the cluster need have a GPU device present. On the nodes that do have one or more NVIDIA GPUs, install the drivers from NVIDIA's repository.
+
+[source,bash]
+----
+# zypper addrepo --refresh https://download.nvidia.com/suse/sle15sp2/ nvidia
+# zypper refresh
+# zypper install x11-video-nvidiaG05
+----
+
+[NOTE]
+====
+For most modern NVIDIA GPUs, the G05 driver will support your device. Some older devices may require the x11-video-nvidiaG04 driver package instead. Check NVIDIA's documentation for your GPU device model.
+====
+
+=== Install the OCI hooks
+
+OCI hooks are a way for vendors or projects to inject executable actions into the lifecycle of a container managed by the container runtime (runc). SUSE provides an OCI hook for NVIDIA that enable the container runtime and therefor the kubelet and the Kubernetes scheduler to query the host system for the presence of a GPU device and access it directly. Install the hook on the worker nodes with GPUs:
+
+[source,bash]
+----
+# zypper install nvidia-container-toolkit
+----
+
+=== Test a GPU container image
+
+At this point, you should be able to run a container image that requires a GPU and directly access the device from the running container, for example using podman:
+
+[source,bash]
+----
+# podman run docker.io/nvidia/cuda nvidia-smi
+----
+
+=== Troubleshooting
+
+At this point, you should be able to run a container image using a GPU. If that is not working, check the following:
+
+Ensure your GPU is visible from the host system:
+
+[source,bash]
+----
+# lspci | grep -i nvidia
+# nvidia-smi
+----
+
+Ensure the kernel modules are loaded:
+
+[source,bash]
+----
+# lsmod | grep nvidia
+----
+
+If they are not, try loading them explicitly and check dmesg for an error indicating why they are missing:
+
+[source,bash]
+----
+# nvidia-modprobe
+# dmesg | tail
+----
+
+== Configure Kubernetes
+
+=== Install the device plugin
+
+The Kubernetes device plugin framework allows the kubelet to advertise system hardware resources that the Kubernetes scheduler can then use as hints to schedule workloads that require such devices. The Kubernetes device plugin from NVIDIA allows the kubelet to advertise NVIDIA GPUs it finds present on the worker node. Install the device plugin using kubectl:
+
+[source,bash]
+----
+$ kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/1.0.0-beta6/nvidia-device-plugin.yml
+----
+
+=== Taint GPU workers
+
+In a heterogeneous cluster, it may be preferable to prevent scheduling pods that do not require a GPU on nodes with a GPU in order to ensure that GPU workloads are not competing for time on the hardware they need to run. To accomplish this, add a taint to the worker nodes that have GPUs:
+
+[source,bash]
+----
+$ kubectl taint nodes worker0 nvidia.com/gpu=:NoSchedule
+----
+
```

Comment:
if we add nodes label would be a good idea in a heterogeneous cluster.
So that we can check which one is GPU enabled nodes by checking labels. 
```
kubectl label nodes <your-node-name> hardware-type=NVIDIAGPU
kubectl get nodes -l hardware-type=NVIDIAGPU
kubectl taint nodes nvidia.com/gpu=:NoSchedule -l hardware-type=NVIDIAGPU
```

Reply:
I think that's a choice the admin could make it they want to organize their cluster that way but it doesn't same them any steps here since they still need to individually edit each node.